provisioner/registryv1/README.md

```diff
@@ -0,0 +1,123 @@
+
+Simply create a `Bundle` resource pointing to a specific version of your bundle, and a `BundleInstance` which references that bundle.
+The registryv1 provisioner will unpack the provided Bundle onto the cluster, and eventually make the content available on the cluster. 
+
+```yaml
+apiVersion: core.rukpak.io/v1alpha1
+kind: Bundle
+metadata:
+  name: my-bundle
+spec:
+  image: my-bundle-image@sha256:xyz123
+  provisionerClassName: core.rukpak.io/registry+v1
+---
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleInstance
+metadata:
+  name: my-bundle-instance
+spec:
+  provisionerClassName: core.rukpak.io/registry+v1
+  bundleName: my-bundle
+``` 
+### Make bundle content available but do not install it
+
+There is a natural separation between sourcing of the content and application of that content via two separate RukPak APIs, `Bundle` and `BundleInstance`. 
+A user can specify a particular `Bundle` to be available in the cluster for inspection before any application of the resources. 
+Given a `Bundle` resource named `my-bundle`, the registryv1 provisioner will pull down and unpack the bundle to a variety of storage backends, such as ConfigMaps or PersistentVolumes.
+
+Currently the ConfigMap backend is supported by the provisioner, and the content can be inspected by viewing the metadata ConfigMap, which holds
+information on all ConfigMaps related to this bundle. 
+
+`kubectl get cm bundle-metadata-my-bundle -o yaml` provides access to this metadata ConfigMap, which then links to all other ConfigMap objects
+where the unpacked data is stored. 
+
+Surfacing the content of a bundle in a more user-friendly way, via a plugin or additional API, is on the RukPak roadmap. 
+
+### Pivot between bundle versions 
+
+The `BundleInstance` API is meant to indicate the version of the bundle that should be active within the cluster. Given two unpacked bundles in the cluster,
+`my-bundle-v0.0.1` and `my-bundle-v0.0.2`, the `spec.bundleName` field of the related `BundleInstance` can be updated from say, the older version,
+to the newer version, and the registryv1 provisioner will be able to successfully create the new content on-cluster and remove the old content.  
+
+## Running locally 
+
+To experiment with the registryv1 provisioner locally, first setup a local cluster. It's recommended to have OLM installed on the cluster as well, 
+which can be done easily by heading to the [releases](https://github.com/operator-framework/operator-lifecycle-manager/releases) section and installing via the install script. 
+
+Once the cluster is setup and OLM is running, take the following steps from the root of the repository:
+* Create the rukpak namespace via `kubectl create ns rukpak-system` by default
+* Run `make bin/registryv1` to create the registryv1 provisioner binary locally
+* Apply the RukPak core CRDs via `kubectl apply -f manifests/`
```

Comment:
Alternatively they can use the `make install` directive but that currently also installs the plain-k8s provisioner manifests which are not useful in this context

Reply:
This probably points to just adding a `make install-registryv1` target (for each other provisioner as well.

Reply:
Yup, sounds good 👍

Reply:
In SDK projects, there is:
1. `make install` (just for APIs) 
2. `make deploy` (APIs and controllers from pre-built docker images, controller has configured RBAC)
3. `make run` (just `go run` the main.go, controller has whatever the developer perms are)
4. `make undeploy (delete everything from `make deploy`)
5. `make uninstall` (delete everything from `make install`)

My typical development flow has two main cycles:
1. Update APIs, run `make install` 
2. Update controllers, run `make run`

The third cycle, which I use much more rarely is when I'm changing RBAC for the controller. In that case, it is necessary to build the images and use `make deploy` so that the controller uses a service account tied to that RBAC.

Does that structure (perhaps with better names or tweaks since we have multiple mains) seem reasonable?

Also, have we considered a single main that runs all of our built-in provisioners? We'd get some memory optimization through shared caches.

Reply:
That seems like a nice setup -- I know we want to cleanup our Makefile a bit once we setup the core provisioners and this seems like a good structure to follow. Once we actually have deployment manifests for a working provisioner it should be easier to wire all this up. 

> Also, have we considered a single main that runs all of our built-in provisioners? We'd get some memory optimization through shared caches.

That's interesting, I can think of pros and cons to that approach. I think the original idea was for everything to be separate (but encourage shared code use), but maybe worth rethinking.